rpcs3/Emu/Cell/SPUThread.cpp

```diff
@@ -3387,31 +3414,86 @@ bool spu_thread::process_mfc_cmd()
 			last_faddr = 0;
 		}
 
-		if (addr == raddr && !g_use_rtm && g_cfg.core.spu_getllar_polling_detection && rtime == vm::reservation_acquire(addr) && cmp_rdata(rdata, data))
+		if (raddr)
 		{
-			// Spinning, might as well yield cpu resources
-			std::this_thread::yield();
+			if (raddr != addr)
+			{
+				// Last check for event before we replace the reservation with a new one
+				if (reservation_check(raddr, rdata))
+				{
+					set_events(SPU_EVENT_LR);
+				}
+			}
+			else
+			{
+				// Check if we can reuse our existing reservation
+				if (rtime == vm::reservation_acquire(addr) && cmp_rdata(rdata, data))
```

Comment:
This seems like we can skip some steps later if you go with reservation_try_lock + cmp_rdata for the optimistic path.

Reply:
I'm not sure I umderstand the motivation behind this, the reads are relaxed and locking can make other threads lose their reservation.

Reply:
I mean, in the case of success, you're still trampling rdata which will still make the other threads lose their reservation.
With try_lock you have the benefit of skipping a second atomic_load + cmp_rdata. In the worst case the outcome should be very similar with the only difference between them being relaxed compare vs atomic cas fail.

Reply:
cmp_rdata is just not cheap enough to avoid the cas

Reply:
Second cmp_rdata is needed regardless of locking when using accurate getllar, this is a special way to load any amount of data atomically on any CPU I think.

Reply:
Ok, got it.